Source/DafnyCore/Backends/CSharp/CsharpCodeGenerator.cs

```diff
@@ -265,15 +265,41 @@ protected override ConcreteSyntaxTree CreateStaticMain(IClassWriter cw, string a
       return wr.NewBlock($"public static void _StaticMain(Dafny.ISequence<Dafny.ISequence<{CharTypeName}>> {argsParameterName})");
     }
 
+    /// <summary>
+    /// Compute the name of the class to use to translate a data-type
+    /// </summary>
+    private string dataTypeName(DatatypeDecl dt) {
+      var protectedName = IdName(dt);
+      if (dt.EnclosingModuleDefinition is not null) {
```

Comment:
```suggestion
      if (dt.EnclosingModuleDefinition is { Name: var moduleName } && moduleName == protectedName) {
```
And then you can directly return. That simplifies the code a bit while increasing the readability.

Reply:
Thanks I was not aware of this syntax from C#